hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -1447,11 +1457,11 @@ protected void tryUpgrade(HoodieTableMetaClient metaClient, Option<String> insta
         tableServiceClient.rollbackFailedWrites(pendingRollbacks, true, true);
       }
 
-      new UpgradeDowngrade(metaClient, config, context, upgradeDowngradeHelper)
-          .run(HoodieTableVersion.current(), instantTime.orElse(null));
-
       metaClient.reloadActiveTimeline();
```

Comment:
Can we just move the timeline reload out of the `tryUpgrade` method and do it outside, so that we can eliminate the tricks of this atomic reference get/set.

Reply:
Updated now, please check. I have made `tryUpgrade` return a boolean and if upgrade happened then reload the meta client outside tryUpgrade but still within the transaction.

Reply:
Add a new method `reloadTableConfig` to address the problem.